statsd/statsd.go

```diff
@@ -196,7 +196,7 @@ type Client struct {
 	// tags are global tags to be added to every statsd call
 	tags           []string
 	flushTime      time.Duration
-	metrics        *ClientMetrics
+	metrics        *statsdTelemetry
```

Comment:
nit: I think we'd have some more consistency if `metrics` were `telemetry` and `telemetry` were `telemetryCli` or something. Feel free to ignore.